src/k8s/cmd/k8s/k8s_config.go

```diff
@@ -2,21 +2,40 @@ package k8s
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/canonical/k8s/cmd/k8s/errors"
+	"github.com/canonical/k8s/pkg/snap"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
 	"github.com/spf13/cobra"
 )
 
+var (
+	configCmdOpts struct {
+		server string
+	}
+)
+
 func newKubeConfigCmd() *cobra.Command {
-	return &cobra.Command{
-		Use:     "config",
+	configCmd := &cobra.Command{
+		Use:     "config --server <server>",
 		Short:   "Generate a kubeconfig that can be used to access the Kubernetes cluster",
 		Hidden:  true,
 		PreRunE: chainPreRunHooks(hookSetupClient),
 		RunE: func(cmd *cobra.Command, args []string) (err error) {
 			defer errors.Transform(&err, nil)
+			snap := snap.NewSnap(os.Getenv("SNAP"), os.Getenv("SNAP_COMMON"))
+
+			isWorker, err := snaputil.IsWorker(snap)
+			if err != nil {
+				return fmt.Errorf("failed to check if node is a worker: %w", err)
+			}
+
+			if isWorker {
+				return fmt.Errorf("this action is restricted on workers")
+			}
```

Comment:
This is not needed here, it's already handled in https://github.com/canonical/k8s-snap/blob/73c17ef31708078874ba4a45f5c599a705c3458e/src/k8s/pkg/k8sd/api/endpoints.go#L46-L50